pkg/volume/gcepd/attacher.go

```diff
@@ -138,6 +141,52 @@ func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Sp
 	return volumesAttachedCheck, nil
 }
 
+
+// search Windows disk number by LUN
+func getDiskID(pdName string, exec mount.Exec) (string, error) {
+	cmd := `Get-GcePdName | select Name, DeviceId | ConvertTo-Json`
+	output, err := exec.Run("powershell", "/c", cmd)
+	if err != nil {
+		klog.Errorf("Get-GcePdName failed, error: %v, output: %q", err, string(output))
+		err = errors.New(err.Error() + " " + string(output))
+		return "", err
+	}
+
+	var data []map[string]interface{}
+	if err = json.Unmarshal(output, &data); err != nil {
+		klog.Errorf("Get-Disk output is not a json array, output: %q", string(output))
+		return "", err
+	}
+
+	for _, pd := range data {
+		if jsonName, ok := pd["Name"]; ok {
+			if name, ok := jsonName.(string); ok {
+				klog.Infof("data name %q", name)
+				if name == pdName {
+					klog.Infof("found the disk %q", name)
+					if diskNum, ok := pd["DeviceId"]; ok {
+						switch v := diskNum.(type) {
+						case int:
+							return strconv.Itoa(v), nil
+						case float64:
+							return strconv.Itoa(int(v)), nil
+						case string:
+							return v, nil
+						default:
+							// diskNum isn't one of the types above
+							klog.V(4).Infof("Disk %q found, but disknumber (%q) is not in one of the recongnized type", name, diskNum)
+						}
+						klog.Warningf("Disk %q found, but could not get disk number(%q)", name, diskNum)
```

Comment:
I think you only need to printing the same message once. 

How often does this code gets exercised for, say, one bad PD without the number?